### ISSUE TITLE
DietPi-Globals | G_AG*: Add apt-fast support

### ISSUE THREAD

**Issue by mzramna** [MichaIng/DietPi#2817]
#### Is your feature request related to a problem? Please describe:

- apt is very slow when you update a system

#### Describe the solution you'd like:

- so it could use apt-fast to speed up this,apt-fast is an apt but with the modification to download using aria2c,and this could download multiple dependencies at same time,and is faster than the official apt-get,apt or aptitude

**Comment by MichaIng** [comment]
@mzramna 
Many thanks for your request.

We did have this in the past and re-validated some time ago, and results was that `apt-fast` is not faster, but slower + includes the overhead of Aria2 etc...: https://github.com/MichaIng/DietPi/issues/1987#issuecomment-410004392

Could you show some (of your own) test results that prove it to be significantly faster?

**Comment by mzramna** [comment]
well,i've tried this into mine pcs,not the pi,it just have any speedup when you do an system upgrade,or install any aplication with multiple dependencies/multiple aplications at same time.
the apt-fast made diference JUST when you parallelize the download,like download 10 or 20 files download at same time,lass then this....it is realy useless

**Comment by mzramna** [comment]
i could do this tests,but...i really dont know how to record this time diference...could you show me how to do it?

**Comment by MichaIng** [comment]
@mzramna
Use the `download` APT command to download a larger number of packages to the current dir and use the `time` command to measure the time:
```
cd /mnt/dietpi_userdata
time apt-get download openjdk*
rm openjdk*
time apt-fast download openjdk*
rm openjdk*
```

**Comment by MichaIng** [comment]
I mark this as closed. I opened an issue on apt-fast GitHub to ask for benchmarks (and get notifications in case of reply). Did some more testing there and could still very hardly find any benefit. Basically it depends on the chosen APT mirror. I found some that are not able to use my full bandwidth, where apt-fast then was slightly faster sometimes. But im most cases it was slower, sometimes dramatically, especially after choosing the best local APT mirror for apt-get.

Also apt-fast. to have any benefit, requires some custom configuration to find multiple fast mirrors. There is some tool that should test then and list the 10 fastest, but it does not show correct result, but slower mirrors in my case. Not sure if it's ping vs bandwidth, however all this effort and overhead still makes me believe that it is not worth it.

But leaned something about Aria2:
- It is able to also download single files from multiple mirrors, downloading and merging chunks.
- So the benefit (if found) is independent from the amount of packages being downloaded.
